docs/guides/caching.md

```diff
@@ -0,0 +1,119 @@
+# Caching
+
+When using the recommended approach for initializing experiences with this SDK, the logic will fetch the experience entry, the data source (bound entries & assets), and used patterns under the hood and return the final experience object.
+
+To inject a manual caching layer in between, the SDK allows you to fetch the content on your own.
+
+## Typical Fetching & Rendering
+
+The simple setup would usually look something like this calling one of the sophisticated fetcher functions `useFetchBySlug`. While the final experience object exposes all fetched content, it doesn't provide a way to inject already fetched content but will fetch everything from the API at once.
+
+```ts
+const { experience, error, isLoading } = useFetchBySlug({
+  slug,
+  localeCode: locale,
+  client,
+  experienceTypeId,
+});
+```
+
+## Caching & Injecting Data
+
+To cache the loaded entities and inject them manually, a custom fetcher function needs to be written that uses `fetchExperienceEntry`, `fetchReferencedEntities`, and `createExperience` which are exposed from the SDK as well.
+
+### Fetching Data
+
+```ts
+function customFetchAllEntities({ client, experienceTypeId, slug, localeCode }) {
+  // Fetch the experience entry including its used patterns (resolved links in fields.usedComponents)
+  const experienceEntry = await fetchExperienceEntry({
+    client,
```

Comment:
I've been trying this snippet with the `withAllLocales` modifier and I get a type error in this line:

<img width="879" alt="Screenshot 2025-04-03 at 17 21 12" src="https://github.com/user-attachments/assets/4ffb8a4e-7f38-4aea-bb5c-0382a8bd6f53" />

We probably need to adjust the types for the `client` but I am not sure if that's gonna have any other side-effects. If I understand correctly the response format with the modifier is different than the plain one, is that right?

Reply:
Thanks for calling out. I resolved this by using a UNION type on the params. I'm also using this type now in the types to ensure that this is supported by the types.